frontend/src/plugins/layout/CarouselPlugin.tsx

```diff
@@ -99,6 +104,15 @@ const CarouselComponent = ({
         return (
           <SwiperSlide key={index}>
             <div
+              onKeyDown={(e) => {
+                // If the target is from a marimo element, stop propagation
+                if (
+                  e.target instanceof HTMLElement &&
+                  e.target.tagName.toLocaleLowerCase().startsWith("marimo-")
+                ) {
+                  e.stopPropagation();
+                }
+              }}
```

Comment:
Could we do this with `onMouseDown` too and then re-enable `simulateTouch=True`?

Reply:
i just tried - got pretty close. it works with sliders, but not with altair, so lets just keep it off